frontend/public/components/graphs/prometheus-graph.tsx

```diff
@@ -18,23 +19,23 @@ export const getPrometheusExpressionBrowserURL = (urls, queries): string => {
   return `${base}/graph?${params.toString()}`;
 };
 
-export const PrometheusGraphLink = connectToURLs(MonitoringRoutes.Prometheus)(
-  ({ children, query, urls }: React.PropsWithChildren<PrometheusGraphLinkProps>) => {
-    const url = getPrometheusExpressionBrowserURL(urls, [query]);
-    return query ? (
-      <a
-        href={url}
-        target="_blank"
-        rel="noopener noreferrer"
-        style={{ color: 'inherit', textDecoration: 'none' }}
-      >
-        {children}
-      </a>
-    ) : (
-      <>{children}</>
-    );
-  },
-);
+const getQueryBrowserURL = (query: string): string => {
+  const params = new URLSearchParams();
+  params.set('query0', query);
+  return `/monitoring/query-browser?${params.toString()}`;
```

Comment:
we should link to `dev` perspective `/metrics` if user does not have access to Monitoring section

Reply:
Thanks @rawagner. I created follow-up PR https://github.com/openshift/console/pull/3541 to handle the dev perspective links.